### ISSUE TITLE
ux bug: bridge needs better error handling when device participant information is lacking

### ISSUE THREAD

**Issue by jzacsh** [mautrix/signal#228]
**tl;dr** for better experience, handle `None` better, and always give a clear AI (_"trust that contact? Look for clearer prompts in Signal or via Matrix send the message `mark-trusted`"_)

As seen in [issue 148](https://github.com/mautrix/signal/issues/148#issuecomment-1005704493) it's possible to get a `None` as the identity of the device. This lead to a lot of confusion for me. I think it's worth fixing this case to at least explain what data _should_ have been there to increase confidence in the direction of the issue (that being a participant in the chat is unidentified that is _not_ you), since often the bridge **itself** can have issues.

Here's the error handling code: https://github.com/mautrix/signal/blob/dd759f99ea13aedb3b5575aaeed56042d7243ab8/mausignald/signald.py#L252-L256

I think this could be as simple as:

```diff
            elif result.get("identityFailure", ""): 
                errors.append(
-                    f"{result['identityFailure']}"
+                    f"{result['identityFailure']}" if f"{result['identityFailure']}" or f"another participant"
                )
```

well not quite[^1], but the illustration is clear hopefully.

So that this  bug doesn't seem contrived, here's me [genuinely confused as an end-user](https://github.com/mautrix/signal/issues/148#issuecomment-1057455455) (moments before I _yet again_ disconnect and resetup signal bot, thinking the bot's messed up). I'm glad work was done to handle this case, but the current handling isn't something I'd expect to work out for most end-users (I don't expect most end-users to go searching for this string on github closed issues as I did).

---

[^1]: I say not quite because I haven't written python in years, so don't trust my patch, but also because I suspect there's something fishy about the API response that this code is handling -- if I was reading this code I'd really want some inline documentation as to why `.get(..., "")` fallback clause is written and then yet there's _further_ logic inside that handles a `None` ... maybe there's a **string** `None` being passed down here? Again, over my head currently...
